File: server-rails-multi/app/policies/label_policy.rb
# frozen_string_literal: true

class LabelPolicy < Rhino::ResourcePolicy
  def permitted_attributes_for_show(user)
    if has_role?(user, "owner") || has_role?(user, "admin") ||
       has_role?(user, "manager") || has_role?(user, "member") ||
       has_role?(user, "viewer")
      ["*"]
    else
      []
    end
  end

  def hidden_attributes_for_show(user)
    []
  end

  def permitted_attributes_for_create(user)
    if has_role?(user, "owner") || has_role?(user, "admin") || has_role?(user, "manager")
      %w[name color]
    else
      []
    end
  end

  def permitted_attributes_for_update(user)
    if has_role?(user, "owner") || has_role?(user, "admin") || has_role?(user, "manager")
      %w[name color]
    else
      []
    end
  end
end

File: server-nestjs-single/src/policies/LabelPolicy.ts
import { ResourcePolicy } from '@rhino-dev/rhino-nestjs';

/**
 * Label policy — SINGLE-TENANT (role-free). Label is a SHARED GLOBAL catalog:
 * no ownership scope, every authenticated user sees all labels. Action
 * authorization comes from User.permissions (`labels.*`).
 */
export class LabelPolicy extends ResourcePolicy {
  override resourceSlug = 'labels';

  override permittedAttributesForShow(): string[] {
    return ['*'];
  }
  override hiddenAttributesForShow(): string[] {
    return [];
  }
  override permittedAttributesForCreate(): string[] {
    return ['name', 'color'];
  }
  override permittedAttributesForUpdate(): string[] {
    return ['name', 'color'];
  }
}
